### ISSUE TITLE
TetrisGram Pacer Test audio falls out of sync

### ISSUE THREAD

**Issue by smiegrin** [joezeng/cambridge#11]
While playing through the Tetrisgram Pacer Test, the level transitions start to fall out of sync with the sounds in the audio. The degree and rate at which this desync happens varies, by my observation, from device to device. My laptop keeps up well into 15 minutes of gameplay, but my desktop only takes about 15 "laps" before it's transitioning a full second before the audio cue.

**Comment by joezeng** [comment]
Oh, it's a device-to-device thing? That's concerning, because it means my formula for calculating time might not be off after all, and it's literally just a problem with the computer dropping frames while audio doesn't drop samples.

**Comment by smiegrin** [comment]
Indeed. I'm pretty new to lua, but from a brief look over the LOVE documentation, there isn't currently a way to find out where the audio is playing at, so I think the sounds are going to have to play manually to keep things in sync. I suspect the opening would still work to preserve a part of the meme-feel, and maybe the beep sounds can still work if they're extracted as dedicated sound effects.

**Comment by joezeng** [comment]
Yeah, beeps would work as dedicated sound effects, since they're presented in isolated form at the beginning of the track.

Although we'd have to find 21 minutes worth of filler background music. I suppose NoCopyrightSounds might have some things worth our while in there.

**Comment by joezeng** [comment]
https://www.ncsmusic.com/usage/

Oh, hmm. Looks like NCS also requires specific licensing to use their music in a game. Phooey.

**Comment by Arcorann** [comment]
For what it's worth, I've confirmed in Audacity that the audio is out by roughly 0.95 seconds compared to the exact timing by the start of level 15, which makes me wonder exactly what's going on. Recording a video and checking the ingame timer against the audio and exact timings might help IMO.

**Comment by joezeng** [comment]
Yeah, I've been meaning to make a spreadsheet to keep track of the calculated and observed start times.

**Comment by ProspectPyxis** [comment]
For background audio, can I suggest anything from [SoundImage](https://soundimage.org/)? Tons of looping music on there, and the only requirement seems to be attribution.
